salt/modules/file.py

```diff
@@ -2014,15 +2014,18 @@ def replace(path,
         pre_group = get_group(path)
         pre_mode = salt.utils.normalize_mode(get_mode(path))
 
-    # Avoid TypeErrors by forcing repl to be a string
-    repl = str(repl)
+    # Avoid TypeErrors by forcing repl to be bytearray related to mmap
+    # Replacement text may contains integer: 123 for example
+    repl = salt.utils.to_bytes(str(repl))
+    if not_found_content:
+        not_found_content = salt.utils.to_bytes(not_found_content)
 
     found = False
     temp_file = None
     content = str(not_found_content) if not_found_content and \
```

Comment:
Won't using ``str(not_found_content)`` here cause an issue if ``not_found_content`` is a bytestring (as you've added in the block above this one)?

```python
>>> x = b'123'
>>> x
b'123'
>>> str(x)
"b'123'"
```